### PR TITLE
Update the stack documentation 

### PR DESCRIPTION
**Pull request by Fumuran**
This tiny PR updates the documentation describing the changes made in the #1456 PR.

Related issue: #1428.

### CODE REVIEW

**Comment by bobbinth** [comment]
I think there are a few more places where we need to make updates. For example, [here](https://0xpolygonmiden.github.io/miden-vm/intro/overview.html#inputs-and-outputs) and [here](https://github.com/0xPolygonMiden/miden-vm/tree/next/miden#inputs--outputs). There may be other places as well - so, let's make sure we update them all.

File: docs/src/intro/overview.md [hunk 1]
```diff
@@ -14,32 +16,37 @@ These components are:
...
 2. Secret (or nondeterministic) inputs can be supplied to the VM via the [*advice provider*](#nondeterministic-inputs). There is no limit on how much data the advice provider can hold.
 
-After a program finishes executing, the elements remaining on the stack become the outputs of the program. Since these outputs will be public inputs for the verifier, having a large stack at the end of execution will increase cost to the verifier. Therefore, it's best to drop unneeded output values. We've provided the [`truncate_stack`](../user_docs/stdlib/sys.md) utility function in the standard library for this purpose.
+After a program finishes executing, the elements remaining on the stack become the outputs of the program. Notice that having more than 16 values on the stack at the end of execution will cause an error, so the values outside the stack top should be dropped. We've provided the [`truncate_stack`](../user_docs/stdlib/sys.md) utility procedure in the standard library for this purpose.
```
Comment (Al-Kindi-0) [inline docs/src/intro/overview.md:31]: nit: I would say ".. so the values beyond the top 16 elements of the stack should be dropped."

File: miden/README.md [hunk 2]
```diff
@@ -21,12 +21,12 @@ All Miden programs can be reduced to a single 32-byte value, called program hash
 Currently, there are 3 ways to get values onto the stack:
 
 1. You can use `push` instruction to push values onto the stack. These values become a part of the program itself, and, therefore, cannot be changed between program executions. You can think of them as constants.
-2. The stack can be initialized to some set of values at the beginning of the program. These inputs are public and must be shared with the verifier for them to verify a proof of the correct execution of a Miden program. While it is possible to initialize the stack with a large number of values, we recommend keeping the number of initial values at 16 or fewer as each initial value beyond 16 increases verifier complexity.
+2. The stack can be initialized to some set of values at the beginning of the program. These inputs are public and must be shared with the verifier for them to verify a proof of the correct execution of a Miden program. At most 16 values could be provided for the stack initialization, attempt to provide more than 16 values will cause an error.
```
Comment (Al-Kindi-0) [inline miden/README.md:24]: nit: attempt -> attempts

File: miden/README.md [hunk 3]
```diff
@@ -21,12 +21,12 @@ All Miden programs can be reduced to a single 32-byte value, called program hash
...
 
 The stack is provided to Miden VM via `StackInputs` struct. These are public inputs of the execution, and should also be provided to the verifier. The secret inputs for the program are provided via the `Host` interface. The default implementation of the host relies on in-memory advice provider (`MemAdviceProvider`) that can be commonly used for operations that won't require persistence.
 
-Values remaining on the stack after a program is executed can be returned as stack outputs. You can specify exactly how many values (from the top of the stack) should be returned. Similar to stack inputs, a large number of values can be returned via the stack, however, we recommend keeping this number to under 16 not to overburden the verifier.
+Values remaining on the stack after a program is executed can be returned as stack outputs. You can specify exactly how many values (from the top of the stack) should be returned. Notice, that, similar to stack inputs, at most 16 values can be returned via the stack. Attempt to return more than 16 values will cause an error.
```
Comment (Al-Kindi-0) [inline miden/README.md:29]: nit: attempt -> attempts

**Review by Al-Kindi-0 — APPROVED** [review]
Looks good, thank you!
Only a few minor nits.

File: docs/src/intro/overview.md (hunk 1 above)
Comment (Fumuran) [reply docs/src/intro/overview.md:31]: I had a feeling that it could be phrased much better, but I couldn't find the right words. Thank you!

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**